### DISCUSSION TITLE
How to connect Victron Smart Solar MPPT and Smart Shunt to OpenDTU-onBattery

### DISCUSSION THREAD

**Discussion by hwoit** [hoylabs/OpenDTU-OnBattery#540] [Q&A] [answered]
Hi there,

I wonder how both a Victron Smart Solar MPPT such as the MPPT 150/35 and a Victron Smart Shunt are connected to the OpenDTU.

There is only the illustration on the wiki‘s main page under the examples section showing them both to be connected to the OpenDTU, but I can not find any hint how both Victron deviced are connected and the pins are assigned correctly.

Looking through the documentation I could only find 1 serial port assignment for Victron Ve.direct and it seems to me, that you have to connect these devices separately.

Does anyone know how to connect both Smart Solar and Smart Shunt correctly?

Thanks in advance!

Holger
![IMG_2620](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/152097286/273f2d32-01ba-471d-bc16-61d022e1c04f)

**Comment by swingstate** [comment]
Did you read the Ve.Direct documentation [here](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface)?

I could also offer you one of these plug n play [devices](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/477#discussioncomment-7660963), together with the 2 necessary cables.

**Reply by Manos1966** [reply]
Yeah, I have put something but as it has not been vetted properly, I hesitate to place it in a more obvious position....
[Plan](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU#relais-no-normally-open--nc-normally-closed--highlow-level-trigger)
@swingstate Any suggestions where to place it when it is ready?

**Reply by swingstate** [reply]
Are you referring to updating the Ve.Direct page in the wiki?

**Reply by Manos1966** [reply]
well, it is more than only VE.Direkt, this is all PIN connections I could find during the past few months.
I put it there when I was experimenting with the Huawei Rectifier and had to work out the pin assignement.
[https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU#relais-no-normally-open--nc-normally-closed--highlow-level-trigger](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU#relais-no-normally-open--nc-normally-closed--highlow-level-trigger)

The big dissadvantage we have is that, if someone has an issue and raises a question, after it has been resolved, it is very difficult for **beginners** to find it when the do a search :( (default searches run on open items only).
This is why I always try to "dump" solutions and comments inside the Wiki, because even if it is unstructured for the time being, we have at least all relevant points somewhere.

**Reply by swingstate** [reply]
ok, so if your question is to update the wiki based on findings from the discussion then yes, I agree :) 
If I am not getting your point, then please help me. Catched a cold and my brain is acting kinda slow.

**Reply by Manos1966** [reply, marked as answer]
Gute Besserung!

well, the question is: 
Where in WiKi do you place a graph like this one? (has the pinout of most devices on it)
![Wiring_ESP32_Symbol - Victr1Victr2PylontechHuaweiRelaisCMT2300updRJ45](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/6960d9b8-7f3a-4920-8268-d74505d6b48f)

**Reply by swingstate** [reply]
I would show that probably somewhere at the top and then also on the different subpages where different subtopics in this context are discussed, and then maybe improve the graphics by highlighting what's being discussed in the text.

**Reply by hwoit** [reply]
Ok, thanks for the help!

I was not able to see the wood for the trees ( also ich hab den Wald nicht vor lauter Bäumen gesehen!).

I do not know why I did not realize, that with „battery“ on the device page
([https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles))
the Victron Smart Shunt was ment. I even did not see the information in the table where the Smart Shunt is apparently mentioned. Sorry for that.

@swingstate 
Thanks for the offer for the plug and play device, but I already ordered a NodeMCU and the NRF24 to build the DTU by myself.

@Manos1966 
The graph is a good hint for the inexperienced user. Has it been in the wiki 2 days ago?

kind regards
Holger

**Reply by Manos1966** [reply]
@hwoit No worries, it happened to me as well.
The developpers here (I am not part of this group) are producing updates at twice the speed you can update the Wiki.
Which in turn, is good (better than the opposite) .

The graph is there for a couple of months but, it is very well hidden :D You have to deserve it to find it :D
I hid it in the Wiki on a discussion regarding the 230V rectifiers 

Give it a few weeks and you will be up-to-speed.

**Reply by greymda** [reply]
@swingstate where i can find details about your plug and play devices? cheers

**Reply by swingstate** [reply]
> @swingstate where i can find details about your plug and play devices? cheers

I am not selling them since they are not yet CE certified.

You can find details about the development process and testing when searching for "onbattery PCB" or "onbattery Pro PCB" here in the  discussions.

**Reply by greymda** [reply]
@swingstate understood, looking forward getting one unit.
